test/functional/wallet_signer.py

```diff
@@ -187,5 +200,36 @@ def run_test(self):
         # )
         # self.clear_mock_result(self.nodes[4])
 
+    def test_invalid_signer(self):
+        self.log.debug(f"-signer={self.mock_invalid_signer_path()}")
+
+        # Create new wallets for an external signer.
+        # disable_private_keys and descriptors must be true:
+        assert_raises_rpc_error(-4, "Private keys must be disabled when using an external signer", self.nodes[1].createwallet, wallet_name='not_hww_invalid', disable_private_keys=False, descriptors=True, external_signer=True)
+        if self.is_bdb_compiled():
+            assert_raises_rpc_error(-4, "Descriptor support must be enabled when using an external signer", self.nodes[1].createwallet, wallet_name='not_hww_invalid', disable_private_keys=True, descriptors=False, external_signer=True)
+        else:
+            assert_raises_rpc_error(-4, "Compiled without bdb support (required for legacy wallets)", self.nodes[1].createwallet, wallet_name='not_hww_invalid', disable_private_keys=True, descriptors=False, external_signer=True)
+
+        self.log.info('Test invalid external signer')
+        try:
+            self.nodes[1].createwallet(wallet_name='hww_invalid', disable_private_keys=True, descriptors=True, external_signer=True)
+        except JSONRPCException:
+            pass
+
+        # Flag can't be set afterwards (could be added later for non-blank descriptor based watch-only wallets)
+        self.nodes[1].createwallet(wallet_name='not_hww_invalid', disable_private_keys=True, descriptors=True, external_signer=False)
+        not_hww = self.nodes[1].get_wallet_rpc('not_hww_invalid')
+        assert_raises_rpc_error(-8, "Wallet flag is immutable: external_signer", not_hww.setwalletflag, "external_signer", True)
+
+        # assert_raises_rpc_error(-4, "Multiple signers found, please specify which to use", wallet_name='not_hww', disable_private_keys=True, descriptors=True, external_signer=True)
+
+        # TODO: Handle error thrown by script
+        # self.set_mock_result(self.nodes[1], "2")
+        # assert_raises_rpc_error(-1, 'Unable to parse JSON',
+        #     self.nodes[1].createwallet, wallet_name='not_hww2', disable_private_keys=True, descriptors=True, external_signer=False
+        # )
+        # self.clear_mock_result(self.nodes[1])
```

Comment:
In a5f53c36942bf82c056380df6c15adf0b12190f2 "Check descriptors returned by external signers"

It is not necessary to duplicate these tests.

Reply:
removed in https://github.com/bitcoin/bitcoin/pull/23628/commits/96846ef7817257326d51c4f05047ca46c3f7f3c7